### PR TITLE
Slightly better error for failed tslint.json parses

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Fixes https://github.com/palantir/tslint/issues/1681

`configuration`'s `findConfiguration` now returns a result wrapper with the attempted `path` as well as `error?` and `results?`. `tslint-cli` will log the failure nicely.

### CODE REVIEW

**Comment by JoshuaKGoldberg** [comment]
@nchen63 added

File: test/executable/executableTests.ts [hunk 1]
```diff
@@ -79,6 +79,17 @@ describe("Executable", function() {
                 done();
             });
         });
+
+        it("exits with code 1 if config file is invalid", (done) => {
+            execCli(["-c", "test/config/tslint-invalid.json", "src/tslint.ts"], (err, stdout, stderr) => {
+                assert.isNotNull(err, "process should exit with error");
+                assert.strictEqual(err.code, 1, "error code should be 1");
+
+                assert.include(stderr, "Invalid option for configuration", "stderr should contain notification about invalid json");
```
Comment (nchen63) [inline test/executable/executableTests.ts]: this test isn't exercising the code in this PR. `Invalid option for configuration` is written when the config file is missing. I was expecting `Failed to load <path>: <error message>`

Comment (JoshuaKGoldberg) [inline test/executable/executableTests.ts]: You're right - that was sloppy. Should be fixed.

**Comment by nchen63** [comment]
@JoshuaKGoldberg thanks!

**Pull request merged**